app.js

```diff
@@ -1,35 +1,173 @@
+    console.log(question, 'user correct:', userAns);
+  } while (typeof userAns != 'boolean')
+  if (userAns) {
+    alert(correctResp);
+    correctCount++;
   } else {
-    var tInput = input.toUpperCase().trim();
-    return tInput == 'YES' || tInput == 'Y';
+    alert(wrongResp);
   }
-};
+}
+console.log('correct y/n answers:', correctCount);
 
-/**
-Takes the question to be asked in a prompt, the answer for a response of 'yes', and the answer for a response of 'no'. Asks the question until the 'OK' button is pressed. Prints the user response as a boolean or null to the console. Responds to the user with the given answer in an alert. Returns nothing.
-**/
-var askQuestion = function(question, yesAns, noAns) {
-  var response;
-  do {
-    response = checkInput(prompt(question));
-    console.log(question, response);
-  } while (typeof response != 'boolean')
-  if (response) {
-    alert(yesAns);
+// GUESS THE NUMBER QUESTIONS
+
+// userAns defined above
+
+var favNum = Math.round((Math.random() * 100));
+console.log('my favorite number is:', favNum);
+questionCount++;
+
+var isNotDone = true;
+var message;
+
+for (var guess = 3; guess >= 0 && isNotDone; guess--) {
+
+  userAns = prompt('Guess my favorite number!');
+  console.log('user gueses:',userAns);
+  if (isNaN(userAns)) {
+    alert('That\'s not a number...')
   } else {
-    alert(noAns);
+    userAns = parseInt(userAns);
+    isNotDone = userAns != favNum;
+
+    if (isNotDone) {
+      message = 'No, that number is too ';
+      if (userAns > favNum) {
+        message += 'large.';
+      } else {
+        message += 'small.';
+      }
+      if (guess != 0) {
+        message += '\nOnly ' + guess + ' ';
+        if (guess != 1) {
+          message += 'guesses remain.';
+        } else {
+          message += 'guess remains.';
+        }
+      } else {
+        message += '\nToo bad, all out of guesses.\nIt was ' + favNum + '.';
+      }
+      alert(message);
+
+    } else {
+      alert('Yay! You guessed right!');
+      correctCount++;
+    }
+
+  }
+}
+
+console.log('current correct answers:', correctCount);
+
+// MULTIPLE ANSWER QUESTIONS
+
+// userAns defined above
+// message defined above
+var possibleAns = ['Canada', 'Mexico', 'France', 'England', 'Wales', 'Scotland'];
+questionCount++;
+isNotDone = true; // isNotDone defined above
+
+for (var guess = 6; guess >= 0 && isNotDone; guess--) {
+  if (guess === 6) {
+    userAns = prompt('Can you guess one of the countries I have traveled to?');
+  } else {
+    userAns = prompt('What\'s another guess?');
+  }
+  console.log('user country guess:', userAns);
+  if (userAns != null && userAns != '') {
+    userAns = userAns.toUpperCase().trim();
+    for (var comp = 0; comp < possibleAns.length; comp++) {
+      if (possibleAns[comp].toUpperCase() === userAns) {
+        isNotDone = false;
+        message = 'Yep, you got one.\n I have been to ' + possibleAns[0];
+        for (var add = 1; add < possibleAns.length; add++) {
```

Comment:
var add is defined two times in app.js. because it's in a for loop but not a function, it's available on the global scope